drivers/rail/rail_drv.c

```diff
@@ -0,0 +1,798 @@
+    DEBUG("using 915MHz radio band\n");
+    ret = RAIL_ConfigChannels(dev->rhandle, &_rail_radio_channel_config, NULL);
+#   endif
+
+
+#endif /*  (RAIL_RADIO_BAND == 868) || (RAIL_RADIO_BAND == 915) */
+
+    if (ret != RAIL_STATUS_NO_ERROR) {
+        LOG_ERROR("Can not init rail radio channels - error msg: %s\n", rail_error2str(ret));
+        return -1;
+    }
+
+    // init IEEE802.15.4 acceleration features
+
+    ret = RAIL_IEEE802154_Init(dev->rhandle, &_rail_ieee802154_config);
+    if (ret != RAIL_STATUS_NO_ERROR) {
+        LOG_ERROR("Can not init rail ieee 802.15.4 support - error msg: %s\n", rail_error2str(ret));
+        return -1;
+    }
+
+    // init radio PA
+
+    r = _rail_PA_init(dev);
+
+    if (r != 0) {
+        return r;
+    }
+
+
+    // setup transmitt buffer
+    ret = RAIL_SetTxFifo(dev->rhandle, _transmit_buffer, 0, sizeof(_transmit_buffer));
+
+    // setup PTI
+#if (RAIL_PTI_ENABLED == 1)
+    r = _rail_PTI_init(dev);
+    assert(r == 0);
+#endif
+
+
+    // if pan coord
+    // setpancoord
+
+    // get mac addr from SoC
+    uint32_t tmp = DEVINFO->UNIQUEL;
+    dev->eui.uint64 = byteorder_htonll((uint64_t)((uint64_t)DEVINFO->UNIQUEH << 32) | tmp);
+
+    DEBUG("Node EUI: %02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x\n",
+          dev->eui.uint8[0],
+          dev->eui.uint8[1],
+          dev->eui.uint8[2],
+          dev->eui.uint8[3],
+          dev->eui.uint8[4],
+          dev->eui.uint8[5],
+          dev->eui.uint8[6],
+          dev->eui.uint8[7]);
+
+    // set panid TODO if in dev, than that?
+    DEBUG("Set PanID to 0x%04x\n", RAIL_DEFAULT_PANID);
+
+    netdev->pan = RAIL_DEFAULT_PANID;
+
+    bool bRet = RAIL_IEEE802154_SetPanId(dev->rhandle, RAIL_DEFAULT_PANID, 0);
+    if (bRet != true) {
+        DEBUG("Can not set PAN ID %d\n", RAIL_DEFAULT_PANID);
+    }
+
+    // set short addr
+    DEBUG("Set ShortAddr 0x%04x\n", ntohs(dev->eui.uint16[3].u16));
+
+    memcpy(netdev->short_addr, &dev->eui.uint16[3].u16, 2);
+
+    bRet = RAIL_IEEE802154_SetShortAddress(dev->rhandle, dev->eui.uint16[3].u16, 0);
+    if (bRet != true) {
+        DEBUG("Can not set short addr\n");
+    }
+
+    // set long addr
+    DEBUG("Set LongAddr 0x%08lx%08lx\n", ntohl(dev->eui.uint64.u32[0]), ntohl(dev->eui.uint64.u32[1]));
+
+    memcpy(netdev->long_addr, &dev->eui.uint8, IEEE802154_LONG_ADDRESS_LEN);
+
+    // reversed order, because thats what rails want
+    uint64_t addr_rev = byteorder_ntohll(dev->eui.uint64);
+
+    bRet = RAIL_IEEE802154_SetLongAddress(dev->rhandle, (uint8_t *) &addr_rev, 0);
+    if (bRet != true) {
+        DEBUG("Can not set long addr\n");
+    }
+
+
+    // get transmitt power
+    RAIL_TxPowerLevel_t power_level_tx = RAIL_GetTxPower(dev->rhandle);
+    RAIL_TxPower_t power_tx_ddBm = RAIL_ConvertRawToDbm(dev->rhandle,
+    #if RAIL_RADIO_BAND == 2400
+                                                        RAIL_TX_POWER_MODE_2P4_HP, // 2.4GHZ HighPower, TODO low power?
```

Comment:
What do you mean here with low power?

Reply:
There are different radio power amplifiers, depending on the specific soc (says the API docu for the driver blob). For the low power version another enumerator have to be used (only applies to 2.4GHz).